util/equality/equality_test.go

```diff
@@ -559,3 +560,82 @@ func TestLoadBalancerServiceChanged(t *testing.T) {
 		}
 	}
 }
+
+func TestContourStatusChangedChanged(t *testing.T) {
+	testCases := []struct {
+		description string
+		current     operatorv1alpha1.ContourStatus
+		mutate      func(status *operatorv1alpha1.ContourStatus)
+		expect      bool
+	}{
+		{
+			description: "if nothing changed",
+			current:     operatorv1alpha1.ContourStatus{},
+			mutate:      func(_ *operatorv1alpha1.ContourStatus) {},
+			expect:      false,
+		},
+		{
+			description: "if available contours changed",
+			current:     operatorv1alpha1.ContourStatus{},
+			mutate: func(status *operatorv1alpha1.ContourStatus) {
+				status.AvailableContours = int32(1)
+			},
+			expect: true,
+		},
+		{
+			description: "if available envoys changed",
+			current:     operatorv1alpha1.ContourStatus{},
+			mutate: func(status *operatorv1alpha1.ContourStatus) {
+				status.AvailableEnvoys = int32(1)
+			},
+			expect: true,
+		},
+		{
+			description: "if a condition is added",
+			current:     operatorv1alpha1.ContourStatus{},
+			mutate: func(status *operatorv1alpha1.ContourStatus) {
+				status.Conditions = []metav1.Condition{
+					{
+						Type:    "Available",
+						Status:  metav1.ConditionFalse,
+						Reason:  "Foo",
+						Message: "Bar",
+					},
+				}
+			},
+			expect: true,
+		},
+		{
+			description: "if a condition reason changes",
+			current: operatorv1alpha1.ContourStatus{
+				Conditions: []metav1.Condition{
+					{
+						Type:    "Available",
+						Status:  metav1.ConditionFalse,
+						Reason:  "Foo",
+						Message: "Bar",
+					},
+				},
+			},
+			mutate: func(status *operatorv1alpha1.ContourStatus) {
+				status.Conditions = []metav1.Condition{
+					{
+						Type:    "Available",
+						Status:  metav1.ConditionFalse,
+						Reason:  "Foo",
+						Message: "Changed",
+					},
+				}
+			},
+			expect: true,
+		},
```

Comment:
Since we only have one condition, I was planning to add ^ test case when additional conditions are introduced. I'll leave a TODO in the test code as a future reminder.